src/documentation/runners/direct.md

```diff
@@ -1,9 +1,47 @@
+* enforcing encodability of elements
+* elements are processed in an arbitrary order at all points
+* serialization of user Fns (DoFn, CombineFn, etc.)
+
+Using the direct runner for testing and development helps ensure that pipelines are robust across different Beam runners. In addition, debugging failed runs can be a non-trivial task when a pipeline executes on a remote cluster. Instead, it is often faster and simpler to perform local unit testing on your pipeline code. Unit testing your pipeline locally also allows you to use your preferred local debugging tools.
+
+Here are some resources with information about how to test your pipelines.
+* [Testing Unbounded Pipelines in Apache Beam](https://beam.incubator.apache.org/blog/2016/10/20/test-stream.html) talks about the use of Java classes [PAssert]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/testing/PAssert.html) and [TestStream]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/testing/TestStream.html) to test your pipelines.
+* The [Apache Beam WordCount Example](http://beam.incubator.apache.org/get-started/wordcount-example/) contains an example of logging and testing a pipeline with `PAssert`.
+
+
+## Direct runner prerequisites and setup
+
+You must specify your dependency on the direct runner.
+
+For example, if you are using Maven for development and want to use the SDK for Java with [DirectRunner]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/runners/direct/DirectRunner.html), add the following dependency to your `pom.xml` file:
+
+```
+   <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>beam-runners-direct-java</artifactId>
+      <version>0.3.0-incubating</version>
+      <scope>runtime</scope>
+   </dependency>
+```
+
+## Pipeline options for the direct runner
+
+When executing your pipeline from the command-line, set `runner` to `direct` or `directrunner`. The default values for the pipeline options are generally sufficient.
+
+See the reference documentation for the  <span class="language-java">[DirectOptions]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/runners/direct/DirectOptions.html)</span><span class="language-python">[PipelineOptions](https://github.com/apache/incubator-beam/blob/python-sdk/sdks/python/apache_beam/utils/options.py)</span> interface (and its subinterfaces) for defaults and the complete list of pipeline configuration options.
+
+## Additional information and caveats
+
+Local execution is limited by the memory available in your local environment. It is highly recommended that you run your pipeline with data sets small enough to fit in local memory. You can create a small in-memory data set using a Create transform, or you can use a Read transform to work with small local or remote files.
```

Comment:
Perhaps code formatting for `Create` and `Read` along with Javadoc/Py links to them?